pytse_client/orderbook/common.py

```diff
@@ -0,0 +1,118 @@
+import datetime
+import logging
+import os
+import pandas as pd
+from pathlib import Path
+from pytse_client.ticker import Ticker
+from pytse_client.config import ORDER_BOOK_HIST_PATH
+
+# maximum available depth in tsetmc for historical orderbook
+MAX_DEPTH = 5
+ERROR_MSG = "{date} is not a valid date object. Make sure it was a trade day."
+
+logging.basicConfig(
+    level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s"
+)
+
+ORDERBOOK_HEADER = {
+    "Accept": "application/json, text/plain, */*",
+    "Accept-Language": "en-US,en;q=0.9,fa-IR;q=0.8,fa;q=0.7",
+    "Cache-Control": "no-cache",
+    "Connection": "keep-alive",
+    "DNT": "1",
+    "Pragma": "no-cache",
+    "User-Agent": "Mozilla/5.0 (Linux; Android 6.0; Nexus 5 Build/MRA58N) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Mobile Safari/537.36",
+}
+
+
+keys = {
+    "datetime": "hEven",
+    "refID": "refID",
+    "depth": "number",
+    "bid": "pMeDem",
+    "ask": "pMeOf",
+    "vol_bid": "qTitMeDem",
+    "vol_ask": "qTitMeOf",
+    "num_bid": "zOrdMeDem",
+    "num_ask": "zOrdMeOf",
+}
+
+base_keys = [
+    "bid",
+    "ask",
+    "vol_bid",
+    "vol_ask",
+    "num_bid",
+    "num_ask",
+]
+
+extended_columns = []
+for i in range(1, MAX_DEPTH + 1):
+    columns = [f"{key}_{i}" for key in base_keys]
+    extended_columns.extend(columns)
+
+reversed_keys = {val: key for key, val in keys.items()}
+
+
+def _validate_trade_date(ticker: Ticker, date: datetime.date):
+    return date in ticker.trade_dates
+
+
+def validate_dates(
+        ticker: Ticker,
+        start_date: datetime.date,
+        end_date: datetime.date,
+        ignore_date_validation: bool
+        ):
+    if not ignore_date_validation:
+        if not _validate_trade_date(ticker, start_date):
+            raise Exception(ERROR_MSG.format(date=start_date))
+        if not _validate_trade_date(ticker, end_date):
+            raise Exception(ERROR_MSG.format(date=end_date))
+
+
+def get_valid_dates(ticker: Ticker,
```

Comment:
We have a similar function here, the logic seems to be the same shall we reuse this?
https://github.com/Glyphack/pytse-client/blob/76bf6a4283b9b7bcac3f7ce7d5e385195bfe5eed/pytse_client/ticker/ticker.py#L381-L382

Reply:
I'm also fine with updating that function if it was not enough for your use case. Just having it in one place would be helpful.

Reply:
So, I'm using the same function in `get_valid_dates`, and I didn't mess with `trade_dates` 'cause it's a property of Ticker. But I can create a wrapper function called `get_range_trade_dates` in `Ticker` to handle my use case. Sound good to you? @Glyphack